tests/TestCase/Routing/Middleware/RoutingMiddlewareTest.php

```diff
@@ -427,4 +429,91 @@ public function scopedMiddlewareUrlProvider()
             ['/api/version', ['second', 'last']],
         ];
     }
+
+    /**
+     * Test we store route collection in cache.
+     *
+     * @return void
+     */
+    public function testCacheRoutes()
+    {
+        Configure::write('Router.cache', true);
+        Cache::setConfig('_cake_router_', [
+            'engine' => 'File',
+            'path' => TMP,
+        ]);
+        Router::$initialized = false;
+        $request = ServerRequestFactory::fromGlobals(['REQUEST_URI' => '/articles']);
+        $response = new Response();
+        $next = function ($req, $res) {
+            $routeCollection = Cache::read('routeCollection', '_cake_router_');
+            $this->assertInstanceOf('\Cake\Routing\RouteCollection', $routeCollection);
```

Comment:
You could use `RouteCollection::class` here.

Reply:
:white_check_mark: